server/auth.go

```diff
@@ -803,9 +844,23 @@ func (s *Server) isGatewayAuthorized(c *client) bool {
 	}
 
 	// Check whether TLS map is enabled, otherwise use single user/pass.
-	if opts.Gateway.TLSMap {
-		return checkClientTLSCertSubject(c, func(user string, _ *ldap.DN) (string, bool) {
-			return "", opts.Gateway.Username == user
+	if opts.Gateway.TLSMap || opts.Gateway.TLSImplicitAllow {
+		return checkClientTLSCertSubject(c, func(user string, _ *ldap.DN, isDNSAltName bool) (string, bool) {
+			if user == "" {
```

Comment:
In original code, is that possible that `user` and `Gateway.Username` be empty in TLSMap context? If so, this is changing the behavior. Is that intended?

Reply:
I don't think that user should ever be empty. In that case the CA would have signed a cert with empty fields.
this is more like a guard I added out of habit. remove?

I'm also not sure. if you can even configure an empty user name.

Reply:
I was just asking, because maybe we need this check and it wasn't there before. Actually, looking at isGatewayAuthorized() top of func we already return true if `opts.Gateway.Username == ""`

Reply:
I think this function (outside of tests at least) is always called with a user, so we should not need this test.
But if we think that cert.EmailAddress, etc.. may contain (legally or maliciously) empty values and if that would cause a failure for dnsAltNameLabels(), then you could leave the check.

Reply:
I just guard against something weird coming out of the cert. that said. I realized that it is missing for the cluster